rustup-init.sh

```diff
@@ -356,16 +371,18 @@ get_architecture() {
         fi
     fi
 
+    local _current_exe
     case "$_ostype" in
 
         Android)
             _ostype=linux-android
             ;;
 
         Linux)
-            check_proc
+            get_current_exe
+            _current_exe=$RETVAL
```

Comment:
Any reason not to se

`_current_exe=get_current_exe`

Reply:
@rbtcollins I assume you meant `_current_exe=$(get_current_exe)`?

The `echo "Warning: ..."` calls will interfere with the function's output when it's being used like so. Looking at the surrounding code, it looks like `$RETVAL` trick is used exactly in this case to specify the actual value to be returned. Is there a better way though?

Reply:
@rbtcollins I just had another look of #3350, and I believe I should switch to `1>&2` for this function. See https://github.com/rust-lang/rustup/issues/3350#issuecomment-2158213483 for the direction we're moving towards in general WRT stdout/stderr handling.

Reply:
@rbtcollins Done.